source/common/quic/envoy_quic_server_preferred_address_config_factory.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "envoy/config/typed_config.h"
+#include "envoy/network/address.h"
+#include "envoy/protobuf/message_validator.h"
+
+#include "quiche/quic/platform/api/quic_socket_address.h"
+
+namespace Envoy {
+namespace Quic {
+
+// An interface to provide the server's preferred addresses at runtime.
+class EnvoyQuicServerPreferredAddressConfig {
+public:
+  virtual ~EnvoyQuicServerPreferredAddressConfig() = default;
+
+  /**
+   * @param local_address the configured default listening address.
+   * Returns a pair of the server preferred addresses in form of {IPv4, IPv6}. An uninitialized
+   * address value means no preferred address for that address family.
```

Comment:
Can you add to the comment when this is called? eg at config-load time vs once-per-connection?

Reply:
done